riff

```diff
@@ -23,6 +25,40 @@ Commands:
 EOF
 }
 
+function print_init_usage() {
+cat <<EOF
+
+Initialize the function based on the function source code specified as the filename, using the name
+  and version specified for the function image repository and tag.
+
+Usage:
+
+  riff init -u <useraccount> -n <name> -v <version> -f <source> -l <language> -p <protocol> -i <input-topic> -o <output-topic> [-a <artifact>] [--classname <class-name>] [--push]
+
+Options:
+
+  -u, --useraccount: the Docker user account to be used for the image repository (defaults to current OS username)
+  -n, --name: the name of the function (defaults to the name of the current directory)
+  -v, --version: the version of the function (defaults to 0.0.1)
+  -f, --filename: filename or directory to be used for the function resources, 
+                  if a file is specified then the file's directory will be used 
+                  (defaults to the current directory)
+  -f, --filename: filename, directory, or URL for the resource definition[s] (defaults to the current directory)
+  -l, --language: the language used for the functiuon source
+                  (defaults to filename extension language type or 'shell' if directory specified)
+  -p, --protocol: the protocol to use for function invocations
+                  (defaults to 'stdio' for shell and python, to 'http' for java and node)
+  -i, --input: the name of the input topic (defaults to function name)
+  -o, --output: the name of the output topic (no default, only created if specified)
+  -a, --artifact: path to the function artifact, source code or jar file
+                  (defaults to function name with extension appended based on language: 
+                      '.sh' for shell, '.jar' for java, 'js' for node and '.py' for python)
+  --classname: the fully qualified class name of the Java function class (required for Java functions)
```

Comment:
this is fine for now, but perhaps we want to generalize this (e.g. "handler") or possibly allow it as an optional suffix on the artifact value (after a delimiter perhaps)

Reply:
yeah, not sure what handler options there will be for other languages and also not sure why we require the classname for Java since SCFn should be able to pull it from the Jar manifest "Function-Class" entry.